src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/component/SNSPublisher.kt

```diff
@@ -24,18 +24,22 @@ class SNSPublisher(
   fun publish(referralId: UUID, actor: AuthUser?, event: EventDTO) {
     if (enabled) {
       buildRequestAndPublish(event)
-      telemetryClient.trackEvent(
-        "InterventionsDomainEvent",
-        mapOf(
-          "event" to event.eventType,
-          "referralId" to referralId.toString(),
-          "actorUserId" to actor?.id,
-        ),
-        null
-      )
     } else {
       logger.debug("Event notification was not published due to sns being disabled.")
     }
+    sendCustomEvent(referralId, actor, event.eventType)
+  }
+
+  private fun sendCustomEvent(referralId: UUID, actor: AuthUser?, eventType: String) {
+    telemetryClient.trackEvent(
+      "InterventionsDomainEvent",
+      mapOf(
+        "event" to eventType,
+        "referralId" to referralId.toString(),
+        "actorUserId" to actor?.id,
+      ),
+      null
+    )
```

Comment:
Some considerations:
1. Would it be useful to provide metrics on SNS send success/fail?
2. Maybe put it in a finally block incase of any exceptions?

Reply:
Thanks, Andrew

1. I'm assuming we could assess failures from number of exceptions, which should always be 0
2. As this telemetry is only for awareness (real data comes from our database), it doesn't sound too important to guard from exceptional cases

What do you think?

Reply:
These are just considerations without knowing in detail how these metrics are used...

I see that it's more tied to overall 'event' itself rather than metrics of an SNS event.

It may be hard to tie errors to specific eventTypes if wanting to have alerts/metrics of exactly where it's failing. But presumably this can be derived from the exceptions.

Reply:
Yes, the call stack on each exception should tell us exactly where it's failing. Thanks for the think about this, my feeling is that it's good enough for our purposes now